src/main/java/org/rumbledb/types/ItemType.java

```diff
@@ -125,12 +125,12 @@ public ItemType(String text) {
     }
```

Comment:
I actually have a bigger concern regarding this class in conjunction with the convertClassNameToItemTypeName() method and ItemType(String text) constructor. The manipulation of these string values which appear to be 'magic values' feels a bit scattered across the place. By all means, at the moment, they function correctly. However the fragility of convertClassNameToItemTypeName() as it directly handles classnames, and the duplication of string magic values in this file and in the ItemParser methods make me uneasy.

I am currently investigating the idea of adding the string representations of item types directly to the itemType class as a field. This way, each item type will statically have their String serialization value and these static and final variables can act as a single source of truth across the project for these serialized strings. Furthermore, if we were to place these within a map that serves as a catalogue(potentially with string representations as the keys), the itemType constructor with the string parameter and the convertClassNameToItemTypeName() method can be rendered redundant. What do you think ?

Reply:
You are right, I was keeping it for later but this is now addressed too. This method is gone.

Instead, items can "self-declare" their dynamic type, but this should only be used for error messages as JSONiq does not expose dynamic types other than through instance of tests.

Reply:
Regarding string representation of item types : this already exists (toString() does that). The problem was to display error messages on items.